gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -57,8 +57,11 @@
   // Job configuration file monitor polling interval in milliseconds
   public static final String JOB_CONFIG_FILE_MONITOR_POLLING_INTERVAL_KEY = "jobconf.monitor.interval";
   public static final long DEFAULT_JOB_CONFIG_FILE_MONITOR_POLLING_INTERVAL = 300000;
-  // Directory where all job configuration files are stored
+  // Directory where all job configuration files are stored WHEN ALL confs reside in local FS.
   public static final String JOB_CONFIG_FILE_DIR_KEY = "jobconf.dir";
+
+  // Path where all job configuration files stored
+  public static final String JOB_CONFIG_FILE_GENERAL_PATH_KEY = "jobconf.fullyQualifiedPath" ;
```

Comment:
How about let user specify fully qualified path as value for existing `JOB_CONFIG_FILE_DIR_KEY` 
.. and in places where you mandatorily need fully qualified path (resolving to local FS) call:  
`jobConfDir.toURI().toString()` or `jobConfDir.toURI().toURL().toString()`
(I believe this returns a string like: `file:/user/foo/bar` when `jobConfDir` is `bar` with parent directory `/user/foo` )